ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/LambdaAnalyzer.fs

```diff
@@ -92,6 +92,17 @@ type LambdaAnalyzer() =
 
     let tryCreateWarning (ctor: ILambdaExpr * 'a -> #IHighlighting) (lambda: ILambdaExpr, replacementExpr: 'a as arg) isFSharp6Supported =
         let lambda = lambda.IgnoreParentParens()
+
+        let replacementExprRef = replacementExpr.As<IFSharpExpression>().IgnoreInnerParens().As<IReferenceExpr>()
```

Comment:
`replacementExprRefExpr` or `replacementRefExpr`, please. It's an expression, not a reference.